File: output/copilot/javascript/easy/q7-contains-duplicate.js
/** https://leetcode.com/problems/contains-duplicate */
// Problem: Contains Duplicate

//Problem Description:
//Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

// Constraints:
// 1 <= nums.length <= 10^5
// -10^9 <= nums[i] <= 10^9

//Code Structure:
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const seen = new Set(); // Create a set to track seen numbers

    for (const num of nums) {
        if (seen.has(num)) {
            return true; // If the number is already in the set, return true
        }
        seen.add(num); // Add the number to the set
    }

    return false; // If no duplicates found, return false
    
};
    

File: output/cursor/java/easy/q7-contains-duplicate.java
/** https://leetcode.com/problems/contains-duplicate */
// Problem: Contains Duplicate

//Problem Description:
//Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

// Constraints:
// 1 <= nums.length <= 10^5
// -10^9 <= nums[i] <= 10^9

//Code Structure:
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        
        for (int num : nums) {
            // If we've seen this number before, it's a duplicate
            if (seen.contains(num)) {
                return true;
            }
            // Add the number to our set of seen numbers
            seen.add(num);
        }
        
        // No duplicates found
        return false;
    }
}
    